packages/api/src/middleware/checkAnimalEntities.js

```diff
@@ -0,0 +1,70 @@
+import { Model, transaction } from 'objection';
+import { handleObjectionError } from '../util/errorCodes';
+
+/**
+ * Middleware function to check if the provided animal entities exist and belong to the farm. The IDs must be passed as a comma-separated query string.
+ *
+ * @param {Object} model - The database model for the correct animal entity
+ * @returns {Function} - Express middleware function
+ *
+ * @example
+ * router.delete(
+ *   '/',
+ *   checkScope(['delete:animals']),
+ *   checkAnimalEntities(AnimalModel),
+ *   AnimalController.deleteAnimals(),
+ * );
+ *
+ */
+export function checkAnimalEntities(model) {
+  return async (req, res, next) => {
+    const trx = await transaction.start(Model.knex());
+
+    try {
+      const { farm_id } = req.headers;
+      const { ids } = req.query;
+
+      if (!ids || !ids.length) {
+        await trx.rollback();
+        return res.status(400).send('Must send ids');
+      }
+
+      const idsSet = new Set(ids.split(','));
+
+      // Check that all animals/batches exist and belong to the farm
+      const invalidIds = [];
+
+      for (const id of idsSet) {
+        // For query syntax like ids=,,, which will pass the above check
+        if (!id || isNaN(Number(id))) {
+          await trx.rollback();
+          return res.status(400).send('Must send valid ids');
+        }
+
+        const existingRecord = await model
```

Comment:
I can't help but wonder if there is a way to do this in one query so that a trx is not needed. While unlikely... having two trx (here and then in delete) means the existing record status could (extremely unlikely) change between the time between actions.

Reply:
I was a little worried by the separate transactions too; actually even opening a transaction in middleware seems a little strange!

In this particular case, what you describe should be safe because the only status that could change like that would be `deleted` itself (record farm ownership isn't going to change) and a re-DELETE on a deleted record won't do anything other than bump the updated properties.